### PR TITLE
2021 08 2 extension updates

### PR DESCRIPTION
**Pull request by temandr**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
dms-preview

**Comment by kairu-ms** [comment]
@temandr Please Fix CI issues. I think you need to keep your branch consistent with latest main branch.

**Review by kairu-ms — APPROVED** [review]

**Review by kairu-ms — COMMENTED** [review]
You should also need to update the version in setup.py file

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+                  "-g {rg} "
+                  "--sku-name {skuname} "
+                  "--subnet {vnetid} "
+                  "--tags area=cli env=test"))
+
+        self.cmd(("az dms project show "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-n {pname1}"),
+                 expect_failure=True)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:67]:
```suggestion
                  "-n {pname1}"),
                  expect_failure=True)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+                         JMESPathCheck('type', 'Microsoft.DataMigration/services/projects')]
+        self.cmd(("az dms project create "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-l {lname} "
+                  "-n {pname1} "
+                  "--source-platform SQL "
+                  "--target-platform SQLDB "
+                  "--tags Type=test Cli"),
+                 checks=create_checks)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:86]:
```suggestion
                  "--tags Type=test Cli"),
                  checks=create_checks)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+
+        self.cmd(("az dms project show "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-n {pname1}"),
+                 create_checks)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:92]:
```suggestion
                  "-n {pname1}"),
                  create_checks)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+                            JMESPathCheck('type', 'Microsoft.DataMigration/services/projects')]
+        self.cmd(("az dms project create "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-l {lname} "
+                  "-n {pnamepg} "
+                  "--source-platform PostgreSQL "
+                  "--target-platform AzureDbForPostgreSQL "
+                  "--tags Type=test Cli"),
+                 checks=create_checks_pg)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:112]:
```suggestion
                  "--tags Type=test Cli"),
                  checks=create_checks_pg)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+        self.cmd(("az dms project show "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-n {pnamepg}"),
+                 create_checks_pg)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:117]:
```suggestion
                  "-n {pnamepg}"),
                  create_checks_pg)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+                            JMESPathCheck('type', 'Microsoft.DataMigration/services/projects')]
+        self.cmd(("az dms project create "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-l {lname} "
+                  "-n {pnamemg} "
+                  "--source-platform MongoDb "
+                  "--target-platform MongoDb "
+                  "--tags Type=test Cli"),
+                 checks=create_checks_mg)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:137]:
```suggestion
                  "--tags Type=test Cli"),
                  checks=create_checks_mg)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 7]
```diff
@@ -0,0 +1,484 @@
...
+        self.cmd(("az dms project show "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-n {pnamemg}"),
+                 create_checks_mg)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:142]:
```suggestion
                  "-n {pnamemg}"),
                  create_checks_mg)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+
+        create_checks_notags = [JMESPathCheck('tags', None)]
+        self.cmd(("az dms project create "
+                  "-g {rg} "
+                  "--service-name {sname} "
+                  "-l {lname} "
+                  "-n {pname2} "
+                  "--source-platform SQL "
+                  "--target-platform SQLDB"),
+                 checks=create_checks_notags)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:152]:
```suggestion
                  "--target-platform SQLDB"),
                  checks=create_checks_notags)
```

File: src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py [hunk 9]
```diff
@@ -0,0 +1,484 @@
...
+
+        list_checks = [JMESPathCheck('length(@)', 4),
+                       JMESPathCheck("length([?name == '{}'])".format(project_name1), 1)]
+        self.cmd(("az dms project list "
+                  "-g {rg} "
+                  "--service-name {sname}"),
+                 list_checks)
```
Comment (jsntcy) [inline src/dms-preview/azext_dms/tests/latest/test_service_scenarios.py:159]:
```suggestion
                  "--service-name {sname}"),
                  list_checks)
```

**Review by jsntcy — APPROVED** [review]

**Pull request merged**